### DISCUSSION TITLE
How do I override the base font size?

### DISCUSSION THREAD

**Discussion by smsi80** [tailwindlabs/tailwindcss#6974] [Help]
i want to change the base font size to 1rem=10px how  can i do it?

**Comment by NovusOpus** [comment]
### There's many ways to achieve this, here are some of them:

### 1. Using plain CSS or  `@Apply` with Arbitrary Value. Easiest but not the most convenient.

### Plain CSS Example:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

html {
  font-size: 62.5%;
}
```
This sets the global default font size to 62.5%, (62.5/100)*16 = 10px

Working example for plain CSS: [Tailwind Play](https://play.tailwindcss.com/I6sAKTZxKC?file=css).

### `@Apply` with Arbitrary Value Example
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

html {
  @apply text-[62.5%]
}
```
This sets the global default font size to 62.5%, (62.5/100)*16 = 10px

[Working example:  Tailwind Play](https://play.tailwindcss.com/V4HHGHeJ0t?file=css)

---
### 2. Adding your own [custom utilities to Tailwind’s utilities layer](https://tailwindcss.com/docs/adding-custom-styles#adding-custom-utilities) in your main CSS file:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer utilities {
  .font-base {
    font-size: 62.5%; 
  }
}
```

Now you can use: `font-base` with a 62.5% of the default size of `1rem(16px)`  for most browsers,  `(62.5/100)*16 = 10px`, and with this way you keep the default Tailwind CSS font sizes. 
```html
<div class="w-full min-h-screen grid place-items-center">
  <p class="font-base">
    Some text with (10px)
  </p>
</div>
```
### You can see it in action here: [Tailwind Play](https://play.tailwindcss.com/r8NlZK4DUf)

if you want to apply this size as the global default, add the `font-base` to your `html`, `body` tag or to your root element/component(react or any other framework)

---

### 3. If you want to [edit or replace the values of Tailwind CSS Default font sizes](https://tailwindcss.com/docs/font-size#customizing-your-theme):

```js
module.exports = {
  theme: {
    fontSize: {
      'base': '0.625rem', // 10px
      'lg': '0.75rem', //12px
      'xl': '0.875rem', //14px
      '2xl': '1rem', //16px
    }
  }
}
```
This is going to replace the value of Tailwind default font sizes.

You can see it in action here: [Tailwind Play](https://play.tailwindcss.com/gGNlphoJDj)

**Comment by CedericPrivat** [comment]
If you want to set the base font size to 10px but preserve the relative proportions of tailwind's utility classes, I recently published a Tailwind CSS plugin 'tailwindcss-base-font-size' that allows you to do just that. You can check it out [here](https://github.com/CedericPrivat/tailwindcss-base-font-size).

Give it a try and let me know, if you have any questions! 👍
